From a499f020dc6ad10eef36fe689439806e35cb452f Mon Sep 17 00:00:00 2001
From: "Tong.Chen" <tong.chen@windriver.com>
Date: Fri, 12 Oct 2018 15:58:04 +0800
Subject: [PATCH 01/11] Revert "FogBugz #538499: klockwork: fix potential
 memory leaks"

This reverts commit 4e1035797f7386b23232c7c5064eb145c00cb5af.

Get patches related with stratix10 clock manager code from sdk kernel
v4.18 to update clock manager drvier, so revert patches of old clock
manager drvier.

Signed-off-by: Tong.Chen <tong.chen@windriver.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/socfpga/clk-gate-s10.c   |    5 +----
 drivers/clk/socfpga/clk-periph-s10.c |    1 -
 drivers/clk/socfpga/clk-pll-s10.c    |   11 -----------
 3 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/drivers/clk/socfpga/clk-gate-s10.c b/drivers/clk/socfpga/clk-gate-s10.c
index f02df91..55cbf5c 100644
--- a/drivers/clk/socfpga/clk-gate-s10.c
+++ b/drivers/clk/socfpga/clk-gate-s10.c
@@ -185,7 +185,6 @@ static void __init __socfpga_gate_init(struct device_node *node,
 		if (IS_ERR(socfpga_clk->sys_mgr_base_addr)) {
 			pr_err("%s: failed to find altr,sys-mgr regmap!\n",
 				__func__);
-			kfree(socfpga_clk);
 			return;
 		}
 	}
@@ -209,10 +208,8 @@ static void __init __socfpga_gate_init(struct device_node *node,
 		return;
 	}
 	rc = of_clk_add_provider(node, of_clk_src_simple_get, clk);
-	if (WARN_ON(rc)) {
-		kfree(socfpga_clk);
+	if (WARN_ON(rc))
 		return;
-	}
 }
 
 void __init socfpga_s10_gate_init(struct device_node *node)
diff --git a/drivers/clk/socfpga/clk-periph-s10.c b/drivers/clk/socfpga/clk-periph-s10.c
index 4dea3f9a..3827a89 100644
--- a/drivers/clk/socfpga/clk-periph-s10.c
+++ b/drivers/clk/socfpga/clk-periph-s10.c
@@ -155,7 +155,6 @@ static __init void __socfpga_periph_init(struct device_node *node,
 
 err_clk:
 	clk_unregister(clk);
-	kfree(periph_clk);
 }
 
 void __init socfpga_s10_periph_init(struct device_node *node)
diff --git a/drivers/clk/socfpga/clk-pll-s10.c b/drivers/clk/socfpga/clk-pll-s10.c
index 6b235ed..231c798 100644
--- a/drivers/clk/socfpga/clk-pll-s10.c
+++ b/drivers/clk/socfpga/clk-pll-s10.c
@@ -136,19 +136,8 @@ static struct clk * __init __socfpga_pll_init(struct device_node *node,
 		kfree(pll_clk);
 		return NULL;
 	}
-
 	rc = of_clk_add_provider(node, of_clk_src_simple_get, clk);
-	if (rc < 0) {
-		pr_err("Could not register clock provider for node:%s\n",
-			clk_name);
-		goto err_clk;
-	}
 	return clk;
-
-err_clk:
-	clk_unregister(clk);
-	kfree(pll_clk);
-	return NULL;
 }
 
 void __init socfpga_s10_pll_init(struct device_node *node)
-- 
1.7.9.5

